infrastructure/builder/src/build/monoimage.js

```diff
@@ -147,8 +156,54 @@ const generateMonoimageTasks = ({tasks, baseDir, spec, cfg, cmdOptions}) => {
     },
   });
 
-  // TODO: add hooks push for web-ui
-  // TODO: add entry point for docker run for web-ui
+  hooks.push({
+    name: 'web-ui',
+    build: async (requirements, utils, procs) => {
+      const cacheDir = path.join(workDir, 'ui/cache');
```

Comment:
This can probably benefit from using the same cacheDir as the 'API Services' section, rather than its own cache.  That should save downloading any packages common between the two.